pandas/core/algorithms.py

```diff
@@ -1414,219 +1414,6 @@ def get_indexer(current_indexer, other_indexer):
-    ("object", "object"): algos.take_2d_axis1_object_object,
-    ("bool", "bool"): _view_wrapper(algos.take_2d_axis1_bool_bool, np.uint8, np.uint8),
-    ("bool", "object"): _view_wrapper(algos.take_2d_axis1_bool_object, np.uint8, None),
-    ("datetime64[ns]", "datetime64[ns]"): _view_wrapper(
-        algos.take_2d_axis1_int64_int64, np.int64, np.int64, fill_wrap=np.int64
-    ),
-}
-
-_take_2d_multi_dict = {
-    ("int8", "int8"): algos.take_2d_multi_int8_int8,
-    ("int8", "int32"): algos.take_2d_multi_int8_int32,
-    ("int8", "int64"): algos.take_2d_multi_int8_int64,
-    ("int8", "float64"): algos.take_2d_multi_int8_float64,
-    ("int16", "int16"): algos.take_2d_multi_int16_int16,
-    ("int16", "int32"): algos.take_2d_multi_int16_int32,
-    ("int16", "int64"): algos.take_2d_multi_int16_int64,
-    ("int16", "float64"): algos.take_2d_multi_int16_float64,
-    ("int32", "int32"): algos.take_2d_multi_int32_int32,
-    ("int32", "int64"): algos.take_2d_multi_int32_int64,
-    ("int32", "float64"): algos.take_2d_multi_int32_float64,
-    ("int64", "int64"): algos.take_2d_multi_int64_int64,
-    ("int64", "float64"): algos.take_2d_multi_int64_float64,
-    ("float32", "float32"): algos.take_2d_multi_float32_float32,
-    ("float32", "float64"): algos.take_2d_multi_float32_float64,
-    ("float64", "float64"): algos.take_2d_multi_float64_float64,
-    ("object", "object"): algos.take_2d_multi_object_object,
-    ("bool", "bool"): _view_wrapper(algos.take_2d_multi_bool_bool, np.uint8, np.uint8),
-    ("bool", "object"): _view_wrapper(algos.take_2d_multi_bool_object, np.uint8, None),
-    ("datetime64[ns]", "datetime64[ns]"): _view_wrapper(
-        algos.take_2d_multi_int64_int64, np.int64, np.int64, fill_wrap=np.int64
-    ),
-}
-
-
-def _get_take_nd_function(
-    ndim: int, arr_dtype: np.dtype, out_dtype: np.dtype, axis: int = 0, mask_info=None
-):
-    if ndim <= 2:
-        tup = (arr_dtype.name, out_dtype.name)
-        if ndim == 1:
-            func = _take_1d_dict.get(tup, None)
-        elif ndim == 2:
-            if axis == 0:
-                func = _take_2d_axis0_dict.get(tup, None)
-            else:
-                func = _take_2d_axis1_dict.get(tup, None)
-        if func is not None:
-            return func
-
-        tup = (out_dtype.name, out_dtype.name)
-        if ndim == 1:
-            func = _take_1d_dict.get(tup, None)
-        elif ndim == 2:
-            if axis == 0:
-                func = _take_2d_axis0_dict.get(tup, None)
-            else:
-                func = _take_2d_axis1_dict.get(tup, None)
-        if func is not None:
-            func = _convert_wrapper(func, out_dtype)
-            return func
-
-    def func2(arr, indexer, out, fill_value=np.nan):
-        indexer = ensure_int64(indexer)
-        _take_nd_object(
-            arr, indexer, out, axis=axis, fill_value=fill_value, mask_info=mask_info
-        )
-
-    return func2
-
-
 def take(
```

Comment:
how are we not moving this?

Reply:
its high-level and i dont want to futz with updating imports everywhere else